src/MultipartBody.cs

```diff
@@ -0,0 +1,158 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Kiota.Abstractions.Extensions;
+using Microsoft.Kiota.Abstractions.Serialization;
+
+namespace Microsoft.Kiota.Abstractions;
+
+/// <summary>
+/// Represents a multipart body for a request or a response.
+/// </summary>
+public class MultipartBody : IParsable
+{
+    private readonly Lazy<string> _boundary = new Lazy<string>(() => Guid.NewGuid().ToString("N"));
+    /// <summary>
+    /// The boundary to use for the multipart body.
+    /// </summary>
+    public string Boundary => _boundary.Value;
+    /// <summary>
+    /// The request adapter to use for serialization.
+    /// </summary>
+    public IRequestAdapter? RequestAdapter { get; set; }
```

Comment:
Out of curiosity, is there a reason this is an `IRequestAdapter` instance and not a `ISerializationWriterFactory` instance? That is all we need to pull writers based on content types.

Reply:
interesting suggestion. The thinking here was that request adapter is what holds the configuration. And what we use in the SetContentFromParsable methods. So mostly consistency and also to reduce code gen (no need to go get the property)